app/data_model/questionnaire_store.py.orig

```diff
@@ -0,0 +1,61 @@
+import json
+import logging
+
+from app.data_model.answer_store import AnswerStore
+<<<<<<< HEAD
+from app.questionnaire.location import Location
+from app.storage.storage_factory import get_storage
+=======
+>>>>>>> Refactoring database storage to store user id/ik
+
+logger = logging.getLogger(__name__)
+
+
+class QuestionnaireStore:
+
+    def __init__(self, storage):
+        self.metadata = {}
+        self.answer_store = AnswerStore()
+        self.completed_blocks = []
+        self.storage = storage
+
+        if self.storage.has_questionnaire_state():
+            data = self.storage.get_user_data()
+
+            if 'METADATA' in data:
+                self.metadata = data['METADATA']
+
+            if 'ANSWERS' in data:
+                self.answer_store.answers = data['ANSWERS']
+
+            if 'COMPLETED_BLOCKS' in data:
+                for location_dict in data['COMPLETED_BLOCKS']:
+                    location = Location(
```

Comment:
Could do something like this (making Location just a wrapper around a dict):

```python
class Location:
    def __init__(..., wrap_dict=None):
         if not wrap_dict:
             self.location = { 'group_id': ..., 'block_id': ...}
        else:
             self.location = wrap_dict

    @property
    def group_id(self):
        return self.location.get('group_id')
....
```

Mean this code could be:

```python
self.completed_blocks = [Location(wrap_dict=d) for d in data['COMPLETED_BLOCKS']]
```

This would also mean that each Location object isn't holding copies of the values but actual references to original dict passed in, it's just wrapping it (you may not agree this is better)

Just an idea...